core/src/main/scala/weaponregex/parser/ParserJVM.scala

```diff
@@ -44,6 +48,39 @@ class ParserJVM private[parser] (pattern: String) extends Parser(pattern) {
     Indexed("""\0""" ~ (CharIn("0-3") ~ CharIn("0-7").rep(exactly = 2) | CharIn("0-7").rep(min = 1, max = 2)).!)
       .map { case (loc, octDigits) => MetaChar("0" + octDigits, loc) }
 
+  /** Parse a single literal character that is allowed to be in a character class
+    * @return [[weaponregex.model.regextree.Character]] tree node
+    * @example `"{"`
+    * @see [[weaponregex.parser.Parser.charClassSpecialChars]]
+    */
+  override def charClassCharLiteral[_: P]: P[Character] =
+    Indexed(CharPred(!charClassSpecialChars.contains(_)).! | "&".! ~ !"&")
```

Comment:
Why excluded `&` by added it in `charClassSpecialChars` and included it separately here?

Reply:
Because you need the negative lookahead. It is a valid character inside the character class, but not if it is followed by another `&`. `!"&"` is the negative lookahead.